### PR TITLE
Add information about Python discord

### PR DESCRIPTION
**Pull request by vors**
Python discord is a vibrant python community built on the modern social platform. A lot of people moved from gitter there.

This PR adds link to the discord, over time ~we would like to consolidate all conversations~ it's possible that discord can consolidate conversations from gitter.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 [![Chat at https://gitter.im/python/typing](https://badges.gitter.im/python/typing.svg)](https://gitter.im/python/typing?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
 
+Chat on [Python Discord](https://www.pythondiscord.com/): [#type-hiting channel](https://discord.com/channels/267624335836053506/891788761371906108)
```
Comment (hauntsaninja) [inline README.md:3]:
```suggestion
Chat on [Python Discord](https://www.pythondiscord.com/): [#type-hinting channel](https://discord.com/channels/267624335836053506/891788761371906108)
```

**Comment by vors** [comment]
updated! Added an image. Unfortunately I was not able to find a reliable cdn address for the image, hence vendored it in the docs folder.

**Review by hauntsaninja — COMMENTED** [review]
You could try https://badgen.net/#discord or shields.io or something... Might need to play around with it a little to get it to link to the channel

**Comment by vors** [comment]
@hauntsaninja thank you for the suggestion!

- For https://badgen.net/#discord I can generate the discord link (yay!) but not the link to a channel. I think it's important to put link to the channel in this README.md (although a general discord one would also be good...)
-  shields.io as far as I can tell doesn't support discord.

The badgen is pretty cool -- it shows the number of members, which looks pretty impressive, however I don't like the fact that there is no discord icon or word in the generated badge. So it's kind of confusing. This custom image that I currently have in the PR is at least non-ambigous.

What do you think about merging it with the current poof-man-version of badge?

**Comment by srittau** [comment]
Are any of the "typing" maintainers hanging out at that discord channel? If that's not the case I wouldn't recommend to add the badge. One of the purposes of these badges ist to be able to connect to us, not just to advertise general Python communication channels.

**Comment by vors** [comment]
I think @hauntsaninja @JelleZijlstra hang out there, not sure are they maintainers or not. There are a lot of helpful conversations happening there.

**Comment by srittau** [comment]
Sure, sounds great in that case.

**Comment by vors** [comment]
Updated to use the badger to avoid commiting the Dicsord image logo to the repo.

Here is how it looks like now

image

**Comment by vors** [comment]
ok, another iteration 😆 

Now I moved it all in the place that's after the discussion guidelines.
image

File: README.md [hunk 2]
```diff
@@ -14,6 +12,10 @@ Improvements to the type system should be discussed on the
 mailing list, although the [issues](https://github.com/python/typing/issues) in this
 repository contain some historic discussions.
 
+For conversations that are more suitable to a chat platform, you can use one of the following:
```
Comment (JelleZijlstra) [inline README.md:15]: Can't comment there but we should remove the mention of Gitter on line 8 above.

Comment (vors) [reply README.md:14]: sharp eye! Done.

File: README.md [hunk 3]
```diff
@@ -1,19 +1,20 @@
...
-chat with us on [Gitter](https://gitter.im/python/typing).
+help in our [support forum](https://github.com/python/typing/discussions).
 
 Improvements to the type system should be discussed on the
 [typing-sig](https://mail.python.org/mailman3/lists/typing-sig.python.org/)
 mailing list, although the [issues](https://github.com/python/typing/issues) in this
 repository contain some historic discussions.
 
+For conversations that are more suitable to a chat platform, you can use one of the following:
+- [gitter](https://gitter.im/python/typing?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
```
Comment (hauntsaninja) [inline README.md:15]:
```suggestion
- [gitter](https://gitter.im/python/typing)
```

**Review by hauntsaninja — APPROVED** [review]
Thanks!

**Pull request merged**